lib/graphql/schema.flow.js

```diff
@@ -333,11 +333,22 @@ export type CategorizationType = $Values<typeof CategorizationTypeValues>;
 
 export type CategorizeTransactionForDeclarationResponse = {|
   __typename?: 'CategorizeTransactionForDeclarationResponse',
+  categoryCode?: ?$ElementType<Scalars, 'String'>,
   elsterCode?: ?$ElementType<Scalars, 'String'>,
   category?: ?$ElementType<Scalars, 'String'>,
   date?: ?$ElementType<Scalars, 'String'>,
 |};
 
+export type CategoryGroup = {|
+  __typename?: 'CategoryGroup',
+  categoryCode: $ElementType<Scalars, 'String'>,
+  elsterCode: $ElementType<Scalars, 'String'>,
```

Comment:
❓ Do we need `elsterCode` in this type? I would have expected that when this is used, elster code is no longer needed

Reply:
all of this is automatically generated from backend. Because backend has `elsterCode` in request/response, so it's generated like this.

Reply:
I know this is auto-generated, but my questions is more about what we are setting on the back-end. I am unsure why we have elsterCode inside CategoryGroup. Maybe I got lost.

But the concept behind my question applies to all comments

Reply:
I see. It is like this because in backend, `categoryGroups` and `elsterGroups` share the same type.

```
@Field((type) => [CategoryGroup])
categoryGroups: CategoryGroup[];

// @deprecated - `elsterCode` should no longer be used. Use `categoryCode` instead.
@Field((type) => [CategoryGroup])
elsterGroups: CategoryGroup[];
```

then

```
class CategoryGroup {
  @Field((type) => String)
  categoryCode: string;

  // @deprecated - `elsterCode` should no longer be used. Use `categoryCode` instead.
  @Field((type) => String)
  elsterCode: string;

  @Field((type) => String)
  categoryCodeTranslation: string;

  // @deprecated - `elsterCode` should no longer be used. Use `categoryCode` instead.
  @Field((type) => String)
  elsterCodeTranslation: string;
}
```

Reply:
If you wonder why they share the same type, then my answer would be to make it safe for backward compatibility and easy to clean up in the future.

Reply:
But maybe I'm too cautious. Let me try to make it cleaner.

Reply:
Ah, I see why I did this. It's not just the type, it's the whole calculation behind it, namely `getDeclarationTransactions`.

Reply:
Thanks for taking the time checking